tests/fsharp/typecheck/tests_typecheck.fs

```diff
@@ -0,0 +1,311 @@
+        // call ..\..\single-neg-test.bat neg77
+        // call ..\..\single-neg-test.bat neg76
+        // call ..\..\single-neg-test.bat neg75
+        // call ..\..\single-neg-test.bat neg74
+        // call ..\..\single-neg-test.bat neg73
+        // call ..\..\single-neg-test.bat neg72
+        // call ..\..\single-neg-test.bat neg71
+        // call ..\..\single-neg-test.bat neg70
+        // call ..\..\single-neg-test.bat neg69
+        // call ..\..\single-neg-test.bat neg68
+        // call ..\..\single-neg-test.bat neg67
+        // call ..\..\single-neg-test.bat neg66
+        // call ..\..\single-neg-test.bat neg65
+        // call ..\..\single-neg-test.bat neg64
+        // call ..\..\single-neg-test.bat neg61
+        // call ..\..\single-neg-test.bat neg63
+        // call ..\..\single-neg-test.bat neg62
+        // call ..\..\single-neg-test.bat neg20
+        // call ..\..\single-neg-test.bat neg24
+        // call ..\..\single-neg-test.bat neg32
+        // call ..\..\single-neg-test.bat neg37
+        // call ..\..\single-neg-test.bat neg37_a
+        // call ..\..\single-neg-test.bat neg60
+        // call ..\..\single-neg-test.bat neg59
+        // call ..\..\single-neg-test.bat neg58
+        // call ..\..\single-neg-test.bat neg57
+        // call ..\..\single-neg-test.bat neg56
+        // call ..\..\single-neg-test.bat neg56_a
+        // call ..\..\single-neg-test.bat neg56_b
+        // call ..\..\single-neg-test.bat neg55
+        // call ..\..\single-neg-test.bat neg54
+        // call ..\..\single-neg-test.bat neg53
+        // call ..\..\single-neg-test.bat neg52
+        // call ..\..\single-neg-test.bat neg51
+        // call ..\..\single-neg-test.bat neg50
+        // call ..\..\single-neg-test.bat neg49
+        // call ..\..\single-neg-test.bat neg48
+        // call ..\..\single-neg-test.bat neg47
+        // call ..\..\single-neg-test.bat neg46
+        // call ..\..\single-neg-test.bat neg10
+        // call ..\..\single-neg-test.bat neg10_a
+        // call ..\..\single-neg-test.bat neg45
+        // call ..\..\single-neg-test.bat neg44
+        // call ..\..\single-neg-test.bat neg43
+        // call ..\..\single-neg-test.bat neg38
+        // call ..\..\single-neg-test.bat neg39
+        // call ..\..\single-neg-test.bat neg40
+        // call ..\..\single-neg-test.bat neg41
+        // call ..\..\single-neg-test.bat neg42
+        do! processor.For (["neg87"; "neg86"; "neg85"; "neg84"; "neg83"; "neg82"; "neg81"; "neg80"; "neg79"; "neg78"; "neg77"; "neg76"; "neg75"; "neg74"; "neg73"; "neg72"; "neg71"; "neg70"; "neg69"; "neg68"; "neg67"; "neg66"; "neg65"; "neg64"; "neg61"; "neg63"; "neg62"; "neg20"; "neg24"; "neg32"; "neg37"; "neg37_a"; "neg60"; "neg59"; "neg58"; "neg57"; "neg56"; "neg56_a"; "neg56_b"; "neg55"; "neg54"; "neg53"; "neg52"; "neg51"; "neg50"; "neg49"; "neg48"; "neg47"; "neg46"; "neg10"; "neg10_a"; "neg45"; "neg44"; "neg43"; "neg38"; "neg39"; "neg40"; "neg41"; "neg42"], singleNegTest)
```

Comment:
because my implementation doesn't work if i try to write as for loop. 
Dont know why. That was something i wanted to ask for help